components/notebook-controller/Dockerfile.konflux

```diff
@@ -26,7 +25,7 @@ WORKDIR /opt/rhods/odh-notebook-controller
 USER root
 
 # Build
-RUN CGO_ENABLED=1 GOOS=${TARGETOS:-linux} GOARCH=${TARGETARCH} GO111MODULE=on go build -tags strictfipsruntime -a -o ./bin/manager main.go
+RUN CGO_ENABLED=1 GOOS=${TARGETOS:-linux} GOARCH=${TARGETARCH} GO111MODULE=on go build -tags strictfipsruntime -a -mod=mod -o manager main.go
```

Comment:
> `-mod=mod`

why is that necessary to specify? isn't `-mod=readonly` (the default) a more reasonable setting?

docs: https://golang.org/ref/mod#build-commands